test/imagebuilder/build.sh

```diff
@@ -55,28 +55,33 @@ while [ "$1" != "" ]; do
     shift
 done
 
-BASE_DIR=/ml
+BASE_DIR=/image_builder
+mkdir $BASE_DIR
+cd $BASE_DIR
 
-echo "Clone ML pipeline code in COMMIT SHA ${COMMIT_SHA}..."
-git clone https://github.com/kubeflow/pipelines ${BASE_DIR}
-cd ${BASE_DIR}
-git config --local user.name 'K8S Bootstrap'
-git config --local user.email k8s_bootstrap@localhost
-git merge --no-ff ${COMMIT_SHA} -m "Merged PR ${COMMIT_SHA}"
+echo "Downloading Docker build context from $CONTEXT_GCS_URI..."
+downloaded_code_archive_file=$(mktemp)
+gsutil cp "$CONTEXT_GCS_URI" "$downloaded_code_archive_file"
+tar -xzf "$downloaded_code_archive_file" --directory .
 
-echo "Waiting for dind to start..."
+echo "Waiting for Docker-in-Docker daemon to start..."
 until docker ps; do sleep 3; done;
 
+gcloud auth configure-docker
+
 if [ "$BUILD_SCRIPT" == "" ]; then
   echo "Build image ${IMAGE_NAME} using ${BASE_DIR}/${DOCKER_PATH}/${DOCKER_FILE}..."
-  docker build -t ${IMAGE_NAME} -f ${BASE_DIR}/${DOCKER_PATH}/${DOCKER_FILE} ${BASE_DIR}/${DOCKER_PATH}
+  docker build -t ${IMAGE_NAME} -f ${BASE_DIR}/${DOCKER_PATH}/${DOCKER_FILE} ${DOCKER_PATH}
 else
   echo "Build image ${IMAGE_NAME} using ${BUILD_SCRIPT}..."
   cd $(dirname ${BUILD_SCRIPT})
-  gcloud auth configure-docker
   bash $(basename ${BUILD_SCRIPT}) -i ${IMAGE_NAME}
 fi
 
-echo "Push image ${IMAGE_NAME} to gcr..."
-gcloud auth configure-docker
+echo "Pushing image ${IMAGE_NAME}..."
 docker push ${IMAGE_NAME}
+
+image_name_with_digest=$(docker inspect --format="{{index .RepoDigests 0}}" "$IMAGE_NAME")
+strict_image_name_output_file=/outputs/strict-image-name/file
+mkdir -p "$(dirname "$strict_image_name_output_file")"
+echo $image_name_with_digest > "$strict_image_name_output_file"
```

Comment:
add comment on what the above four lines do?

Reply:
These lines are for future developments. I want the image builder to output the strict image name (containing digest) that points to the exact image it built. This will help us use the correct image even if for some reason the image builder was asked to build two images with same name, but different content.

Reply:
Added the comment.